File: public/actions/python/createPolicy.py
from cloudproof_py.cover_crypt import Policy, PolicyAxis


async def create_policy():
    """Create Policy"""
    policy = Policy()
    
    policy.add_axis(
        PolicyAxis(
            "department",
            [
                ("Marketing", False),
                ("HR", False),
            ],
            hierarchical=False,  # this is NOT a hierarchical axis
        )
    )
    policy.add_axis(PolicyAxis(
            "country",[
                ("France", False), 
                ("Spain", False), 
                ("Germany", False)],
            hierarchical=False,  # this is NOT a hierarchical axis
        )
    )

    return policy

File: public/actions/javascript/createPolicy.ts
import { CoverCrypt, Policy } from "cloudproof_js";

//
// Creating a Policy
//

type PolicyAxisItem = {
  [key: string]: { name: string; isHybridized: boolean }[];
};

export const createPolicy = async (axis: PolicyAxisItem[]): Promise<Policy> => {
  const { Policy, PolicyAxis } = await CoverCrypt();
  const policyAxis = axis.map((entry) => {
    return new PolicyAxis(Object.keys(entry)[0], Object.values(entry)[0], false);
  });
  const policy = new Policy(policyAxis);
  return policy;
};
